packages/stacked_generator/lib/src/generators/getit/stacked_locator_content_generator.dart

```diff
@@ -1,35 +1,30 @@
 import 'package:stacked_generator/src/generators/base_generator.dart';
-import 'package:stacked_generator/src/generators/enums/dependency_type.dart';
-import 'package:stacked_generator/src/generators/getit/dependency_config.dart';
-import 'package:stacked_generator/src/generators/getit/services_config.dart';
-import 'package:stacked_generator/utils.dart';
+import 'package:stacked_generator/src/generators/getit/dependency_config/factory_param_dependency.dart';
+
+import 'dependency_config/dependency_config.dart';
 
 class StackedLocatorContentGenerator extends BaseGenerator {
   final String locatorName;
   final String locatorSetupName;
-  final ServicesConfig servicesConfig;
+  final List<DependencyConfig> dependencies;
 
   StackedLocatorContentGenerator({
-    required this.servicesConfig,
+    required this.dependencies,
     required this.locatorName,
     required this.locatorSetupName,
   });
-
+  @override
   String generate() {
-    final services = servicesConfig.services;
     writeLine("// ignore_for_file: public_member_api_docs");
 
-    _generateImports(services);
+    _generateImports(dependencies);
 
     newLine();
     writeLine('final $locatorName = StackedLocator.instance;');
     newLine();
 
-    final hasPresolve = services
-        .any((service) => service.type == DependencyType.PresolvedSingleton);
-
     writeLine(
-        '${hasPresolve ? 'Future' : 'void'} $locatorSetupName ({String? environment , EnvironmentFilter? environmentFilter}) ${hasPresolve ? 'async' : ''} {');
+        'Future<void> $locatorSetupName ({String? environment , EnvironmentFilter? environmentFilter}) async {');
```

Comment:
If we have no presolve then this doesn't need to be a Future. Which is why I had the original check there. 

If it's always a Future that means the main function will always have to be turned into a Future?

Reply:
I think it's more consistent, and there is no harm of doing it

Reply:
@eb-so we'll have to update the stacked readme to direct users to always update the main from `void` to `Future<void>` and call `await setupLocator()` instead of only `setupLocator`

Reply:
I'll make a ticket for that